ToolBox/AlgoSeekOptionsConverter/AlgoSeekOptionsConverter.cs

```diff
@@ -70,35 +72,43 @@ public AlgoSeekOptionsConverter(Resolution resolution, DateTime referenceDate, s
         {
             //Get the list of all the files, then for each file open a separate streamer.
-            var files = Directory.EnumerateFiles(_remote, _remoteMask);
-            Log.Trace("AlgoSeekOptionsConverter.Convert(): Loading {0} AlgoSeekOptionsReader for {1} ", files.Count(), _referenceDate);
+            var compressedRawDatafiles = Directory.EnumerateFiles(_remote, _remoteMask).Select(f => new FileInfo(f)).ToList();
+
+            Log.Trace("AlgoSeekOptionsConverter.Convert(): Loading {0} AlgoSeekOptionsReader for {1} ", compressedRawDatafiles.Count(), _referenceDate);
 
             //Initialize parameters
-            var totalLinesProcessed = 0L;
-            var totalFiles = files.Count();
+            var totalLinesProcessed = 0L;   
+            var totalFiles = compressedRawDatafiles.Count();
             var totalFilesProcessed = 0;
             var start = DateTime.MinValue;
 
-            var zipper = OS.IsWindows ? "C:/Program Files/7-Zip/7z.exe" : "7z";
             var random = new Random((int)DateTime.Now.Ticks);
 
-            //Extract each file massively in parallel.
-            Parallel.ForEach(files, parallelOptionsProcessing, file =>
+            //Process each file massively in parallel.
+            Parallel.ForEach(compressedRawDatafiles, parallelOptionsProcessing, compressedRawDatafile =>
             {
-                Log.Trace("Remote File :" + file);
-
-                var csvFile = Path.Combine(_source, Path.GetFileName(file).Replace(".bz2", ""));
-
-                Log.Trace("Source File :" + csvFile);
-
-                if (!File.Exists(csvFile))
+                var timer = DateTime.Now;
+                var rawDataFile = new FileInfo(Path.Combine(_source, compressedRawDatafile.Name.Replace(".bz2", "")));
+                var decompressSuccessful = false;
+                var counter = 1;
+                do
                 {
-                    Log.Trace("AlgoSeekOptionsConverter.Convert(): Extracting " + file);
+                    var attempt = counter == 1 ? string.Empty : $" attempt {counter} of 3";
+                    Log.Trace($"AlgoSeekOptionsConverter.Convert(): Extracting {compressedRawDatafile.Name}{attempt}.");
+                    decompressSuccessful = DecompressOpraFile(compressedRawDatafile, rawDataFile);
+                    counter++;
+                } while (!decompressSuccessful && counter <= 3);
 
-                    var cmdArgs = " e " + file + " -o" + _source;
-                    RunZipper(zipper, cmdArgs);
+                if (!decompressSuccessful)
+                {
+                    Log.Error($"Error decompressing {compressedRawDatafile}. Process Stop.");
+                    throw new NotImplementedException();
                 }
 
+                Log.Trace($"AlgoSeekOptionsConverter.Convert(): Extraction successful in {DateTime.Now - timer:g}, deleting {compressedRawDatafile.Name}.");
+                File.Delete(compressedRawDatafile.FullName);
```

Comment:
info - can also be written as `compressedRawDatafile.Delete()` (See [`FileInfo.Delete`](https://msdn.microsoft.com/en-us/library/system.io.fileinfo.delete%28v=vs.110%29.aspx?f=255&MSPPError=-2147217396))